gossip/resolver.go

```diff
@@ -25,38 +25,62 @@ import (
 	"github.com/cockroachdb/cockroach/util/log"
 )
 
-// Resolver represents the different types of node resolvers.
-// Based on the ResolverType, it will return the same
+// Resolver is an interface which provides an abstract factory for
+// net.Addr addresses.
+type Resolver interface {
```

Comment:
I kept away from an interface since it wasn't quite needed yet, but my thought was to have each type of resolver implement the interface (eg: one for unix, one for static ip/dns, and yet another for load balancer). Not particularly critical though.